core/modules/css/KvInterfaceImplMySql.cc

```diff
@@ -0,0 +1,503 @@
+     * @return true if an int result could be obtained else false
+     */
+    bool extractIntValueFromSqlResults(sql::SqlResults* results, unsigned int* value) {
+        // ideally results would be a sql::SqlResults const*, but extractFirstValue is non-const
+        if (NULL == results || NULL == value) {
+            throw CssError("null inout variable");
+        }
+
+        std::string resStr;
+        sql::SqlErrorObject errObj;
+        results->extractFirstValue(resStr, errObj);
+        if (errObj.isSet()) {
+            return false;
+        }
+        try {
+            *value = boost::lexical_cast<unsigned int>(resStr);
+        } catch (boost::bad_lexical_cast& ex) {
+            return false;
+        }
+        return true;
+    }
+}
+
+// todo this is copied more or less directly from QMetaTransaction. Refactor to shared location?
+// proper place is to port it into db module
+// https://github.com/LSST/db
+// right now it's python only.
+// create a ticket to make generalized code to be used between c++ and python
+class KvTransaction
+{
+public:
+    // Constructors
+    KvTransaction(sql::SqlConnection& conn)
+        : _errObj(), _trans(conn, _errObj) {
+        if (_errObj.isSet()) {
+            throw CssError(_errObj);
+        }
+    }
+
+    // Destructor
+    ~KvTransaction() {
+        // instead of just destroying SqlTransaction instance we call abort and see
+        // if error happens. We cannot throw here but we can print a message.
+        if (_trans.isActive()) {
+            _trans.abort(_errObj);
+            if (_errObj.isSet()) {
+                LOGF(_logger, LOG_LVL_ERROR, "Failed to abort transaction: mysql error: (%1%) %2%" %
+                     _errObj.errNo() % _errObj.errMsg());
+            }
+        }
+    }
+
+    /// Explicitly commit transaction, throws SqlError for errors.
+    void commit() {
+        _trans.commit(_errObj);
+        if (_errObj.isSet()) {
+            LOGF(_logger, LOG_LVL_ERROR, "Failed to commit transaction: mysql error: (%1%) %2%" %
+                 _errObj.errNo() % _errObj.errMsg());
+            throw CssError(_errObj);
+        }
+    }
+
+    /// Explicitly abort transaction, throws SqlError for errors.
+    void abort() {
+        _trans.abort(_errObj);
+        if (_errObj.isSet()) {
+            LOGF(_logger, LOG_LVL_ERROR, "Failed to abort transaction: mysql error: (%1%) %2%" %
+                 _errObj.errNo() % _errObj.errMsg());
+            throw CssError(_errObj);
+        }
+    }
+
+private:
+    sql::SqlErrorObject _errObj; // this must be declared before _trans
+    sql::SqlTransaction _trans;
+};
+
+
+KvInterfaceImplMySql::KvInterfaceImplMySql(mysql::MySqlConfig const& mysqlConf)
+: _conn(mysqlConf) {
+
+}
+
+
+void
+KvInterfaceImplMySql::_findParentId(std::string const& childKvKey, bool* hasParent, unsigned int* parentKvId) {
+
+    // This should only be called when a transaction is in process.
```

Comment:
another nitpick - one way to ensure that transaction has started is to require it (transaction object) as a parameter to this method (even if you do not do anything with that object).